src/Sarif.Viewer.VisualStudio.Core/ErrorList/ErrorListService.cs

```diff
@@ -62,7 +62,10 @@ static ErrorListService()
 
         public static void ProcessLogFile(string filePath, string toolFormat, bool promptOnLogConversions, bool cleanErrors, bool openInEditor)
         {
-            ThreadHelper.JoinableTaskFactory.Run(() => ProcessLogFileWrapperAsync(filePath, toolFormat, promptOnLogConversions, cleanErrors, openInEditor));
+            if (!IsSarifLogOpened(filePath))
```

Comment:
>IsSarifLogOpened [](http://example.com/codeflow?start=17&length=16)

should be the caller's response to check if the SarifLog is already opened? what if the caller want to re-open the save Sarif Log file after its updated?

Reply:
This is a bug fix, but you may be right that we don't necessarily want to do this.
The bug happens when I write the log file to the .sarif folder. The file watcher's Created event fires four times which results in four copies of each result in the error list.
The refactor I did in this change exposed this problem by changing the event sequence.
I'll see about a less brutal solution.

Reply:
Weird. Once I moved the two event handlers out of SarifViewerPackage, this problem stopped happening. I wonder if it has something to do with IVsTaskList2. Or maybe it was some kind of bad state. Blech.

Reply:
good to see its resolved!